plugin/pkg/admission/priority/admission.go

```diff
@@ -211,12 +212,14 @@ func (p *priorityPlugin) admitPod(a admission.Attributes) error {
 			}
 
 			priority = pc.Value
+			preempting = pc.Preempting
 		}
 		// if the pod contained a priority that differs from the one computed from the priority class, error
 		if pod.Spec.Priority != nil && *pod.Spec.Priority != priority {
 			return admission.NewForbidden(a, fmt.Errorf("the integer value of priority (%d) must not be provided in pod spec; priority admission controller computed %d from the given PriorityClass name", *pod.Spec.Priority, priority))
 		}
 		pod.Spec.Priority = &priority
+		pod.Spec.Preempting = preempting
```

Comment:
before we stomp here, do the same mismatch check we do for priority to catch pods that explicitly try to set preemption policies that don't match their priority class:
```
if pod.Spec.PreemptionPolicy != nil && *pod.Spec.PreemptionPolicy != preemptionPolicy {
... error
```

Reply:
Done